hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -424,9 +428,15 @@ private RSQLParameterUnsupportedFieldException createRSQLParameterUnsupportedExc
                 transformedValues.add(convertValueIfNecessary(node, fieldName, value, fieldPath));
             }
 
+            this.joinsNeeded = this.joinsNeeded || areJoinsNeeded(node);
+
             return mapToPredicate(node, fieldPath, node.getArguments(), transformedValues, fieldName, finalProperty);
         }
 
+        private boolean areJoinsNeeded(final ComparisonNode node) {
+            return !Lists.newArrayList("!=", "=out=").contains(node.getOperator().getSymbol());
```

Comment:
Isn't a constant better instead of create a list with the same values all the time?